src/rabbit_peer_discovery.erl

```diff
@@ -183,6 +204,34 @@ unregister() ->
       ok
   end.
 
+-spec lock() -> ok | {ok, Data :: term()} | not_supported | {error, Reason :: string()}.
+
+lock() ->
+    Backend = backend(),
+    rabbit_log:info("Will try to lock with peer discovery backend ~s", [Backend]),
+    case Backend:lock(node()) of
```

Comment:
Should the backend indicate whether locking is supported, like we do for registration? Should `lock/1` and `unlock/1` be a part of https://github.com/rabbitmq/rabbitmq-common/blob/rabbitmq-server-1257/src/rabbit_peer_discovery_backend.erl?